lib/prelude.dx

```diff
@@ -1895,3 +1895,14 @@ def assert (b:Bool) : {Except} Unit =
 --   if ordinal i == 0
 --     then Left (toList [1,2,3])
 --     else Right 1
+
+
+-- TODO: For discussion in PR
+-- This instance is orphaned at bottom of file so it can make use of concat.
+-- Can't easily move concat above Show here (without also breaking up the misc utilities section)
+-- Options:
+-- - remove concat usage, drop back to less efficient repeated <>, colocate with other Show instances
+-- - move Show interface to e.g. show.dx, import (and probably re-export to avoid breakages) in prelude
+-- - split out concat+others to a list.dx, again re-export from prelude
+instance [Show a, Show b] Show (a & b)
+  show = \x. concat ["(", show (fst x), ", ", show (snd x), ")"]
```

Comment:
I think that at least for now it should be ok to just use the less efficient `<>` option. There will be time when we'll have to think very carefully about our string processing, but it hasn't come yet IMO :)

Reply:
Sure thing - changed to use <> and moved it back up to the rest of the Show instances